### PR TITLE
Implemented preview functionality

### PR DESCRIPTION
**Pull request by KasperDamgaard**
Cannot test it locally since I am blocked by CORS when calling the egress API. I did, however, test the resulting http request with postman, which gives back the expected json payload. So it should work. I copied this payload to the backend as a dummy endpoint and tested it that way.

### CODE REVIEW

**Comment by KasperDamgaard** [comment]
Table looks like this:
image

and data access pane is altered as such:
image

File: src/app/pages/details-page/details-page.component.ts [hunk 1]
```diff
@@ -123,4 +179,57 @@ export class DetailsPageComponent implements OnInit, OnDestroy {
...
+      date.getFullYear() - yearsToSubtract,
+      date.getMonth() - monthsToSubtract,
+      date.getDate() - daysToSubtract,
+      date.getHours() - hoursToSubtract,
+      date.getMinutes() - minutesToSubtract,
+      date.getSeconds(),
+      date.getMilliseconds()
+    );
+  }
+
```
Comment (MadsSvejstrup) [inline src/app/pages/details-page/details-page.component.ts:234]: I don't like this method - if we add another code then we have to remember to update it here as well. I suggest that we extend the DurationResponse with an integer that is the number of minutes. In the backend we already have a method that can convert the code to a TimeSpan. 

If we do this then this method is simply just subtracting the number of minutes in the DurationResponse

File: src/app/pages/details-page/details-page.component.ts [hunk 2]
```diff
@@ -112,6 +120,54 @@ export class DetailsPageComponent implements OnInit, OnDestroy {
...
+              .subscribe(val => this.messageNotifier.sendMessage(`${val}. Error message:\n${error.message}`, true));
+          } else {
+            this.translator.get('details.side.access.preview.error.generic')
+              .subscribe(val => this.messageNotifier.sendMessage(`${val}. Error message:\n${error.message}`, true));
+          }
+        });
+      } else {
+        this.translator.get('details.side.access.preview.missingToken').subscribe(val => this.messageNotifier.sendMessage(val, true));
+      }
+    });
```
Comment (MadsSvejstrup) [inline src/app/pages/details-page/details-page.component.ts:168]: The logic that fetches data from the egress api should definitely be pulled out of the component and put in a service instead.

**Comment by MadsSvejstrup** [comment]
Shouldn't we just start fetching a token when pressing 'preview data' instead of throwing an error if a token has not already been fetched? Having to press multiple buttons in the correct order does not seem very nice.

**Comment by MadsSvejstrup** [comment]
If i click 'preview data' first and then later press 'get token', then it tries to fetch the preview data when clicking 'preview data' - that's unexpected behavior because that was not the button i pressed

**Review by MadsSvejstrup — CHANGES_REQUESTED** [review]

**Review by MadsSvejstrup — APPROVED** [review]
I think it looks great! 

How does the table behave if there are a lot of columns? Will we be able to scroll sideways then?

**Comment by KasperDamgaard** [comment]
> I think it looks great!
> 
> How does the table behave if there are a lot of columns? Will we be able to scroll sideways then?

No clue - We'll find out :) - I'd rather do a bugfix on that rather than do too much testing. But I think it might just be squeezed together too much.

**Pull request merged**